core/src/test/scala/zio/schema/codec/JsonCodecSpec.scala

```diff
@@ -109,7 +109,7 @@ object JsonCodecSpec extends DefaultRunnableSpec {
         assertEncodes(
           adtSchema,
           Enumeration(StringValue("foo")),
-          JsonCodec.Encoder.charSequenceToByteChunk("""{"value":{"string":"foo"}}""")
+          JsonCodec.Encoder.charSequenceToByteChunk("""{"oneOf":{"StringValue":{"value":"foo"}}}""")
```

Comment:
@jdegoes I'm not 100% sure about this but I think the encoding before was wrong w/r/t case classes with ADT fields. Not sure if the way it was implemented was just an optimization though....